Refactor and modularize user guide (#826)
pydata/pydata-sphinx-theme: docs/user_guide/branding.rst

=================
Branding and logo
=================

Customize logo and title
========================

By default the theme will use the value of ``project`` on the left side of the header navbar.
This can be replaced by a Logo image, and optionally a custom ``html_title`` as well.

Single logo for light and dark mode
-----------------------------------

To use a local image file, put an image in a folder that is in `html_static_path`, and use the following configuration:

.. code:: python

   html_static_path = ["_static"]
   html_logo = "_static/logo.png"

To use an external link to an image, make sure the ``html_logo`` begins with ``http``.
For example:

.. code:: python

   html_logo = "https://pydata.org/wp-content/uploads/2019/06/pydata-logo-final.png"

Different logos for light and dark mode
---------------------------------------

> You may specify use a different version of your logo image for "light" and "dark" modes.

Reviewer: ```suggestion
You may specify a different version of your logo image for "light" and "dark" modes.
```

This is useful if your logo image is not adapted to a dark mode (light background, not enough contrast, etc...).

> To do so, put the 2 image files in a folder that is in ``html_static_path`` and configure the relative path to each image with ``logo.image_light`` and ``logo.image_dark`` in ``html_theme_options``, like so:

Reviewer: ```suggestion
To do so, put the 2 image files in a folder that is in ``html_static_path`` and configure the relative path to each image with ``logo["image_light"]`` and ``logo["image_dark"]`` in ``html_theme_options``, like so:
```

.. code-block:: python

   html_static_path = ["_static"]
   html_theme_options = {
      "logo": {
         "image_light": "logo-light.png",
         "image_dark": "logo-dark.png",
      }
   }

.. note::

>    ``image_light`` and ``image_dark`` will override the ``html_logo`` setting. If you only specify one of them, but not the other, then the un-specified setting will re-use ``html_logo``.

Reviewer: ```suggestion
   ``image_light`` and ``image_dark`` will override the ``html_logo`` setting. If you only specify one of the light or dark variants, the un-specified variant will fall back to the value of ``html_logo``.
```
